integration-cli/docker_api_logs_test.go

```diff
@@ -82,3 +82,86 @@ func (s *DockerSuite) TestLogsAPIContainerNotFound(c *check.C) {
 	c.Assert(err, checker.IsNil)
 	c.Assert(resp.StatusCode, checker.Equals, http.StatusNotFound)
 }
+
+func (s *DockerSuite) TestLogsAPIUntilFutureFollow(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	until := daemonTime(c).Unix() + 5
+	out, _ := dockerCmd(c, "run", "-d", "-t", "busybox", "/bin/sh", "-c", "while true; do echo hello; sleep 1; done")
+	id := strings.TrimSpace(out)
+	c.Assert(waitRun(id), checker.IsNil)
+
+	type logOut struct {
+		out string
+		err error
+	}
+
+	chLog := make(chan logOut)
+	res, body, err := request.Get(fmt.Sprintf("/containers/%s/logs?until=%d&follow=1&stdout=1&timestamps=1", id, until))
+	c.Assert(err, checker.IsNil)
+	c.Assert(res.StatusCode, checker.Equals, http.StatusOK)
+
+	go func() {
+		defer body.Close()
+		out, err := bufio.NewReader(body).ReadString('\n')
+		if err != nil {
+			chLog <- logOut{"", err}
+			return
+		}
+		chLog <- logOut{strings.TrimSpace(out), err}
+	}()
+
+	select {
+	case l := <-chLog:
+		c.Assert(l.err, checker.IsNil)
+
+		t, err := time.Parse(time.RFC3339Nano, strings.Split(l.out, " ")[0])
+		c.Assert(err, checker.IsNil)
+		c.Assert(t.Unix(), checker.LessOrEqualThan, until)
+	case <-time.After(20 * time.Second):
+		c.Fatal("timeout waiting for logs to exit")
+	}
+}
+
+func (s *DockerSuite) TestLogsAPIUntil(c *check.C) {
+	name := "logsuntil"
+	dockerCmd(c, "run", "--name", name, "busybox", "/bin/sh", "-c", "for i in $(seq 1 3); do sleep 1; echo log$i; done")
+
+	extractBody := func(c *check.C, url string) []string {
```

Comment:
`readLogs` and provide a set of query options

This could also use the API client instead and will be a bit simpler.